# HG changeset patch
# User Maxim Dounin <mdounin@mdounin.ru>
# Date 1315324516 -14400
# Node ID 3b75127939ee38809e0847f09de5c139f241053e
# Parent  a5f19d575163e79c38bdf7e67dc22a72ebb3a7dd
Variables: honor no_cacheable for not_found variables.
Variables with not_found flag set follow the same rules as ones with valid
flag set.  Make sure ngx_http_get_flushed_variable() will flush non-cacheable
variables with not_found flag set.
This fixes at least one known problem with $args not available in subrequest
(with args) when there were no args in main request and $args variable was
queried in main request (reported by Laurence Rowe aka elro on irc).
Also this eliminates unneeded call to ngx_http_get_indexed_variable() in
cacheable case (as it will return cached value anyway).
diff --git a/src/http/ngx_http_variables.c b/src/http/ngx_http_variables.c
--- a/src/http/ngx_http_variables.c
+++ b/src/http/ngx_http_variables.c
@@ -427,7 +427,7 @@ ngx_http_get_flushed_variable(ngx_http_r
 
     v = &r->variables[index];
 
-    if (v->valid) {
+    if (v->valid || v->not_found) {
         if (!v->no_cacheable) {
             return v;
         }
